lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
OK no problem. So we're saying that the handler must match the payload and if they don't then expect strangeness.

What is a "reasonable" timeout while waiting for the UUID to come in? And if it doesn't, should we continue as if it was just a "normal" payload?